src/core/lv_refr.c

```diff
@@ -952,6 +952,29 @@ static void refr_obj_matrix(lv_layer_t * layer, lv_obj_t * obj)
     layer->_clip_area = clip_area_ori;
 }
 
+static bool refr_check_obj_clip_overflow(lv_layer_t * layer, lv_obj_t * obj)
+{
+    if(lv_obj_get_style_transform_rotation(obj, 0) == 0) {
```

Comment:
I haven't tested it but it might be required to test `lv_obj_has_flag(obj, LV_OBJ_FLAG_OVERFLOW_VISIBLE)` too.

Reply:
It seems that the determination of multi-level rotation + clipping is a relatively complicated problem. This is the real-time clipping effect in testcase, sometimes correct and sometimes incorrect.

https://github.com/lvgl/lvgl/assets/26767803/e3b2eb90-2d08-47fd-a781-3a1bc71533eb

Reply:
This is to turn off the contrast between `LV_DRAW_TRANSFORM_USE_MATRIX`.

https://github.com/lvgl/lvgl/assets/26767803/c22fc76e-7b25-4202-979c-f9ff4a707c6c

Reply:
@kisvegabor 

I tried adding:  lv_obj_has_flag(obj, LV_OBJ_FLAG_OVERFLOW_VISIBLE) to the `refr_check_obj_clip_overflow` function:

```c
static bool refr_check_obj_clip_overflow(lv_layer_t * layer, lv_obj_t * obj)
{
    if(lv_obj_get_style_transform_rotation(obj, 0) == 0) {
        return false;
    }

    lv_obj_t * parent = lv_obj_get_parent(obj);
    if(parent && lv_obj_has_flag(parent, LV_OBJ_FLAG_OVERFLOW_VISIBLE)) {
        return true;
    }

    ...
}
```

It looks pretty close, but sometimes there's a little bit more around the edges.

Reply:
I'm having a bit of trouble reproducing the "little bit more around the edges". The left button is always correctly clipped for me. What combination of `LV_USE_MATRIX`, `LV_USE_VECTOR_GRAPHIC`, and `LV_DRAW_TRANSFORM_USE_MATRIX` are you using? `1`, `1`, `1`?

FYI when I disable `LV_USE_VECTOR_GRAPHIC` I get this and a crash when I move the mouse over
![image](https://github.com/lvgl/lvgl/assets/30486941/d5304ce7-6c7c-4d30-8bb5-324afeb8cf26)

Reply:
@liamHowatt